packages/vite/src/node/plugins/esbuild.ts

```diff
@@ -348,6 +364,16 @@ export const buildEsbuildPlugin = (config: ResolvedConfig): Plugin => {
 
       return res
     },
+
+    generateBundle(options) {
+      if (collectLegalComments.length && options.dir) {
+        this.emitFile({
+          fileName: 'LEGAL.txt',
```

Comment:
I think we need to use `name: 'LEGAL.txt'` instead so that the file name includes a hash. That requires the `res.code += '\n/*! For license information please see LEGAL.txt */'` to be moved into `generateBundle` hook.